### ISSUE TITLE
TriggerMap: Unable to open file TwoSiteTriggers.trig

### ISSUE THREAD

**Issue by reginacaputo** [zoglauer/megalib#26]
Hi Andreas, 
I'm currently running the experimental branch of MEGAlib (v2.99.04)
I was working through the trigger map example as outlined in the ReadMe file in: 
${MEGALIB}/resource/examples/advanced/TriggerMap

Follow the instructions, however the TriggerMap files don't seem to get created (see output below - note in the snippet below, I replaced my directory with ${MEGALIB}). I'm sure I'm missing something obvious here. Thanks for taking a look. 

Best, 
Regina 
 
>>make -f ${MEGALIB}/resource/standalone/Makefile.StandAlone PRG=CreateTriggerMap
Program CreateTriggerMap successfully compiled...
Program CreateTriggerMap successfully linked... and starting...

>>CreateTriggerMap -g TriggerMap.geo.setup
Accepting file name: TriggerMap.geo.setup
Stage 1 (reading of file(s)) finished after 0.00650593 sec
Stage 2 (evaluating constants) finished after 0.0101271 sec
Stage 3 (evaluating vectors) finished after 0.0104782 sec
Stage 4 (evaluating for loops) finished after 0.0563427 sec
Stage 5 (evaluating random numbers) finished after 0.0566904 sec
Stage 6 (evaluating if clauses + initial maths evaluation) finished after 0.0683399 sec
Stage 7 (analyzing primary keywords) finished after 0.0787211 sec
Stage 8 (analyzing "Copies") finished after 0.0864317 sec
This file does not exist: "${MEGALIB}/resource/examples/advanced/TriggerMap/TwoSiteTriggers.trig"
MParser: Unable to open file "TwoSiteTriggers.trig".
   *** ERROR **** in trigger TwoSiteTriggers
Unable to open file TwoSiteTriggers.trig
   ***  Error  ***  in setup file ${MEGALIB}/resource/examples/advanced/TriggerMap/TriggerMap.geo.setup at line 219:
"TwoSiteTriggers.Map TwoSiteTriggers.trig"
Unable to read trigger map
Stopping to scan geometry file!
Unable to load geometry  - Aborting!
Error during analysis!

**Comment by zoglauer** [comment]
Hi Regina, 

Fixed the problem - we had a little chicken and egg problem, so I split the geometry in two files, Base.geo.setup contains the volumes and detectors. Use this to generate the trigger maps. TriggerMap.geo.setup then includes everything including the newly generated trigger map. 

Thus do:
CreateTriggerMap -g Base.geo.setup
and from this point on you can use the TriggerMap.geo.setup geometry.

The new version has been pushed to github.

Cheers,
Andreas

**Comment by reginacaputo** [comment]
Thank you, Andreas. I got and compiled the code. It created the trigger files. I then tried to run Cosima on TriggerMap.source. I got an error that it couldn't parse the number of events that I were fed into TriggerMapLow (see below). I tried changing the number, and checking if there were any hidden characters that were messing things up and couldn't find anything. Any ideas? 

Thank you, 
Regina

>>make -f ${MEGALIB}/resource/standalone/Makefile.StandAlone PRG=CreateTriggerMap
Program CreateTriggerMap successfully compiled...
Program CreateTriggerMap successfully linked... and starting...

>> CreateTriggerMap -g Base.geo.setup 
Accepting file name: Base.geo.setup
Stage 1 (reading of file(s)) finished after 0.00314062 sec
Stage 2 (evaluating constants) finished after 0.00531227 sec
Stage 3 (evaluating vectors) finished after 0.00561881 sec
Stage 4 (evaluating for loops) finished after 0.0408925 sec
Stage 5 (evaluating random numbers) finished after 0.0414274 sec
Stage 6 (evaluating if clauses + initial maths evaluation) finished after 0.0556753 sec
Stage 7 (analyzing primary keywords) finished after 0.0666838 sec
Stage 8 (analyzing "Copies") finished after 0.0736185 sec
Stage 9 (analyzing all properties) finished after 0.0874711 sec
Stage 10 (generating clones) finished after 0.0878369 sec
   ***  Warning  ***  
You have not defined any trigger criteria!!
Stage 11 (validation & post-processing) finished after 0.301156 sec
Stage 12 (volume tree optimization) finished after 0.301249 sec
Geometry TriggerMapGeometry loaded!
Program exited normally!

>> ls
ACSVeto.trig          Base.geo.setup        Combined.trig         CreateTriggerMap.cxx  GuardRingVeto.trig    OneSiteTriggers.trig  PairTrigger.trig      ReadMe.txt            TriggerMap.geo.setup  TriggerMap.source     TwoSiteTriggers.trig

>>cosima TriggerMap.source 
Using parameter file TriggerMap.source

   ***  Error  ***  in parsed file /Users/rcaputo2/MEGAlib/resource/examples/advanced/TriggerMap/TriggerMap.source at line 18:
   ***  Error  ***  Cannot parse token Events correctly: Number not a positive (long) integer.
   Line (3 tokens): "TriggerMapLow.Events 				100000"

MParser: Unable to parse file "TriggerMap.source"
Error: Unable to parse parameter file "TriggerMap.source" correctly!
